python/tests/test_genotypes.py

```diff
@@ -2100,3 +2102,60 @@ def test_variant_frequencies_no_samples(self, ts_fixture, caplog):
             freqs = variant.frequencies()
             assert caplog.text.count("frequencies undefined") == 1
         assert np.all(np.isnan(list(freqs.values())))
+
+    def test_variant_str(self):
+        """
+        Test using a simple dummy tree sequence for testing.
+        It has only one tree and one site, whose variant has the alleles
+        ('A', 'T', 'G', None).
+        """
+        tables = tskit.TableCollection(10)
+        tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE, time=0)
+        tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE, time=0)
+        tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE, time=0)
+        tables.sites.add_row(position=5, ancestral_state="A")
+        tables.mutations.add_row(site=0, node=0, derived_state="T")
+        tables.mutations.add_row(site=0, node=1, derived_state="G")
+        ts = tables.tree_sequence()
+        v = next(ts.variants())
+        assert v.alleles == ("A", "T", "G", None)
+        assert isinstance(str(v), str)
+        assert re.match(
+            textwrap.dedent(
+                r"""
+                ╔═+╗
+                ║Variant\s*║
+                ╠═+╤═+╣
+                ║Site id\s*│\s*0║
+                ╟─+┼─+╢
+                ║Site position\s*│\s*[0-9]+\.0║
+                ╟─+┼─+╢
+                ║Number of samples\s*│\s*[0-9]+║
+                ╟─+┼─+╢
+                ║Number of alleles\s*│\s*[0-9]+║
+                ╟─+┼─+╢
+                ║Samples with allele \'A\'\s*│\s*[0-9]+\s*\([0-9\.]+\%\)║
+                ╟─+┼─+╢
+                ║Samples with allele \'T\'\s*│\s*[0-9]+\s*\([0-9\.]+\%\)║
+                ╟─+┼─+╢
+                ║Samples with allele \'G\'\s*│\s*[0-9]+\s*\([0-9\.]+\%\)║
+                ╟─+┼─+╢
+                ║Samples with allele \'missing\'\s*│\s*[0-9]+\s*\([0-9\.]+\%\)║
+                ╟─+┼─+╢
+                ║Has missing data\s*│\s*True║
+                ╟─+┼─+╢
+                ║Isolated as missing\s*│\s*True║
+                ╚═+╧═+╝
+                """[
+                    1:
+                ]
+            ),
+            str(v),
+        )
+
```

Comment:
Can you add in a check for no samples and also when the variant isn't set to a site?

```python
def test_variant_str_no_samples():
    v = next(ts_fixture.variants(samples=[]))
    for allele in v.alleles:
        if allele is not None:
            assert re.search(f"║Samples with allele '{allele}'\s*│\s*0\s*\(nan\%\)║", str(v)) is not None

def test_variant_str_no_site():
   v = tskit.Variant(ts_fixture)
   s = str(v)
   assert len(s.splitlines()) == 3
   assert "This variant has not yet been decoded at a specific site, call Variant.decode to set the site" in s


```